database/src/pool/postgres.rs

```diff
@@ -1525,6 +1525,42 @@ where
 
         Ok(())
     }
+
+    async fn attach_shas_to_try_benchmark_request(
+        &mut self,
+        pr: u32,
+        sha: &str,
+        parent_sha: &str,
+    ) -> anyhow::Result<()> {
+        let tx = self
+            .conn_mut()
+            .transaction()
+            .await
+            .context("failed to start transaction")?;
+
+        tx.execute(
+            "UPDATE
+                benchmark_request
+            SET
+                tag = $1,
+                parent_sha = $2,
+                status = $3
+            WHERE
+                pr = $4
+                AND commit_type = 'try';",
```

Comment:
```suggestion
                AND commit_type = 'try'
                AND tag IS NULL;",
```
We need to only modify try builds on this PR that are not completed and don't have a SHA already. We might also add `WHERE status = "waiting-for-artifscts"` for good measure.

Please add a test for this, that when you have a completed try request in the DB, and then you create another one waiting for artifacts and attach SHA to it, it won't overwrite the SHA of the previous one.

Reply:
I've used the enum to lower the number of hardcoded strings, but yes makes sense